jetty-security/src/main/java/org/eclipse/jetty/security/HashLoginService.java

```diff
@@ -208,8 +218,8 @@ protected void doStart() throws Exception
     protected void doStop() throws Exception
     {
         super.doStop();
-        if (_propertyUserStore != null)
-            _propertyUserStore.stop();
-        _propertyUserStore = null;
+        if (_userStore != null)
+            _userStore.stop();
+        _userStore = null;
```

Comment:
I think we need to remember whether the _userStore was explicitly set or we autocreated it: only if we autocreated it should we null it out, otherwise a stop/restart of the context won't work.